### ISSUE TITLE
Accessibility findings in element section

### ISSUE THREAD

**Issue by marcus-herrmann** [owncloud/owncloud-design-system#18]
First look, incomplete, not yet checked with assistive technology:

- [x] oc-button: Primary button needs a distinct focus style
- [x] oc-checkbox: Needs a focus style, needs a label ([implicit or explicit](https://www.w3.org/WAI/tutorials/forms/labels/))
- [x] oc-image: Alt attribute is missing. Attribute can be left blank if image is purely decorative, but must exist
- [x] oc-radio: Needs a focus style, needs a label ([implicit or explicit](https://www.w3.org/WAI/tutorials/forms/labels/))
- [x] oc-spinner: "spinner" is a rather odd accessible name example in aria-label. Suggestion: "content is loading". Another suggestion would be the use of an aria-live region
- [ ] oc-switch: Needs a focus style, needs a label ([implicit or explicit](https://www.w3.org/WAI/tutorials/forms/labels/))
- [x] oc-textarea: In output HTML the label prop is applied on the textarea. Should be aria-label or an [implicit or explicit label](https://www.w3.org/WAI/tutorials/forms/labels/)
- [ ] oc-textarea: Promote the use of placeholder attribute to supply an example instead of repeating of substituting the label/aria-label: https://www.w3.org/WAI/tutorials/forms/instructions/
- [x] oc-text-field: In output HTML the label prop is applied on the textarea. Should be aria-label or an [implicit or explicit label](https://www.w3.org/WAI/tutorials/forms/labels/)
- [ ] oc-text-field: Promote the use of placeholder attribute to supply an example instead of repeating of substituting the label/aria-label: https://www.w3.org/WAI/tutorials/forms/instructions/

**Comment by PVince81** [comment]
would you be able to work on the items above and provide the missing parts ? @marcus-herrmann

**Comment by marcus-herrmann** [comment]
@PVince81 in parts, yes. But some of the findings are "provide (better) focus styles", and this is design territory or its implementation should be postponed after UX/agreement on final design language, as far as I can tell.

**Comment by pascalwengerter** [comment]
@marcus-herrmann could you update this ticket according to the progress on current `master`? Not sure if the placeholder attribute in forms is still a hard requirement (got different feedback in the workshops AFAIK), and the OcSwitch was intentionally not made accessible since we're not using it in `web`

**Comment by marcus-herrmann** [comment]
@pascalwengerter Can I find the current master at https://web.owncloud.com/? As for the ticket, it is from February 2019 and was referring to a now depracated interface.

Regardless of the current master: a placeholder is in virtually no circumstance a hard requirement in theory, should never _replace_ the label, only supplement the label and maybe serve as an input example.

**Comment by pascalwengerter** [comment]
> @pascalwengerter Can I find the current master at https://web.owncloud.com/? As for the ticket, it is from February 2019 and was referring to a now depracated interface.
> 
> Regardless of the current master: a placeholder is in virtually no circumstance a hard requirement in theory, should never _replace_ the label, only supplement the label and maybe serve as an input example.
> 
> /update: since the unchecked checkboxes only refer to the topics of placeholder and OcSwitch (of February 2019!), and I answered the first and you the second, this seems to be resolved? The ticket lost its fundament, nevertheless, due to the changed interface.

Current ODS master is not deployed publicly, only on an auditing instance (while we're getting the `a11y-swarming` branch merged back into web's `master`). You could check owncloud.design and see the components in action there, but I think we covered the topics here. Closing this for now, thanks everyone for working on this 🙌🏽
